pkg/daemon/daemon.go

```diff
@@ -539,8 +540,25 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 	}
 
 	if reqReboot {
-		glog.Info("nodeStateSyncHandler(): reboot node")
-		rebootNode()
+		if err := wait.PollImmediateUntil(3*time.Second, func() (bool, error) {
+			if utils.ClusterType == utils.ClusterTypeOpenshift {
+				cmc, _ := dn.node.Annotations[mcdconst.CurrentMachineConfigAnnotationKey]
+				dmc, _ := dn.node.Annotations[mcdconst.DesiredMachineConfigAnnotationKey]
+				mcdState, _ := dn.node.Annotations[mcdconst.MachineConfigDaemonStateAnnotationKey]
+				if cmc == "" || cmc != dmc || mcdState != mcdconst.MachineConfigDaemonStateDone {
```

Comment:
@yuqi-zhang @sinnykumari could you review if we are using the MCO conditional check properly?

Reply:
Is it possible, instead of using per node status, we instead use the pool status?
By doing just per node, you risk violating MCP's maxUnavailable, since a separate node could be updating due to the MCO.

Reply:
The per node status checks is not fully equivalent to the MCO, if we want to match the "unavailable" definition of the MCO. Also, you probably need to re-fetch the node object here via an API call, since otherwise you're polling on an old object right?

Reply:
I've changed the logic to check all the nodes of the cluster. So it will wait until MCO finishes all its work across the cluster.

There's a node informer which can guaranty the node object is up-to-date.